<a href="https://colab.research.google.com/github/MariiaSam/data_science/blob/main/Descriptive_Analytics_with_Pandas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [2]:
file_path = '/content/drive/My Drive/sales_data (1).csv'
df = pd.read_csv(file_path)
df.head(5)

Order_Date      Time  Aging  Customer_Id Gender Device_Type  \
0  2018-01-01  10:11:40    5.0        29317   Male         Web   
1  2018-01-01  22:30:44    7.0        42270   Male         Web   
2  2018-01-01  21:55:31   10.0        14563   Male         Web   
3  2018-01-01  13:57:15    9.0        58601   Male         Web   
4  2018-01-01  15:17:41    2.0        48342   Male         Web   

  Customer_Login_type    Product_Category                 Product  Sales  \
0              Member  Auto & Accessories       Car Media Players  140.0   
1              Member  Auto & Accessories  Car Pillow & Neck Rest  231.0   
2              Member  Auto & Accessories            Car Speakers  211.0   
3              Member  Auto & Accessories                    Tyre  250.0   
4              Member  Auto & Accessories                    Tyre  250.0   

   Quantity  Discount  Profit  Shipping_Cost Order_Priority Payment_method  
0       4.0       0.3    43.2            4.3         Medium       e_wallet  
1       5.0       0.1   139.5           13.9           High    money_order  
2       5.0       0.1   120.5           12.0           High    credit_card  
3       4.0       0.2   150.0           15.0       Critical    credit_card  
4       1.0       0.1   165.0           16.5           High    credit_card

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51290 entries, 0 to 51289
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Order_Date           51290 non-null  object 
 1   Time                 51290 non-null  object 
 2   Aging                51289 non-null  float64
 3   Customer_Id          51290 non-null  int64  
 4   Gender               51290 non-null  object 
 5   Device_Type          51290 non-null  object 
 6   Customer_Login_type  51290 non-null  object 
 7   Product_Category     51290 non-null  object 
 8   Product              51290 non-null  object 
 9   Sales                51289 non-null  float64
 10  Quantity             51288 non-null  float64
 11  Discount             51289 non-null  float64
 12  Profit               51290 non-null  float64
 13  Shipping_Cost        51289 non-null  float64
 14  Order_Priority       51288 non-null  object 
 15  Payment_method       51290 non-null 

In [4]:
df.head(5)

Order_Date      Time  Aging  Customer_Id Gender Device_Type  \
0  2018-01-01  10:11:40    5.0        29317   Male         Web   
1  2018-01-01  22:30:44    7.0        42270   Male         Web   
2  2018-01-01  21:55:31   10.0        14563   Male         Web   
3  2018-01-01  13:57:15    9.0        58601   Male         Web   
4  2018-01-01  15:17:41    2.0        48342   Male         Web   

  Customer_Login_type    Product_Category                 Product  Sales  \
0              Member  Auto & Accessories       Car Media Players  140.0   
1              Member  Auto & Accessories  Car Pillow & Neck Rest  231.0   
2              Member  Auto & Accessories            Car Speakers  211.0   
3              Member  Auto & Accessories                    Tyre  250.0   
4              Member  Auto & Accessories                    Tyre  250.0   

   Quantity  Discount  Profit  Shipping_Cost Order_Priority Payment_method  
0       4.0       0.3    43.2            4.3         Medium       e_wallet  
1       5.0       0.1   139.5           13.9           High    money_order  
2       5.0       0.1   120.5           12.0           High    credit_card  
3       4.0       0.2   150.0           15.0       Critical    credit_card  
4       1.0       0.1   165.0           16.5           High    credit_card

# ***Перетворення даних***

In [5]:
df["Order_Date"] = df["Order_Date"] + " " +  df["Time"]
df

Order_Date      Time  Aging  Customer_Id Gender Device_Type  \
0      2018-01-01 10:11:40  10:11:40    5.0        29317   Male         Web   
1      2018-01-01 22:30:44  22:30:44    7.0        42270   Male         Web   
2      2018-01-01 21:55:31  21:55:31   10.0        14563   Male         Web   
3      2018-01-01 13:57:15  13:57:15    9.0        58601   Male         Web   
4      2018-01-01 15:17:41  15:17:41    2.0        48342   Male         Web   
...                    ...       ...    ...          ...    ...         ...   
51285  2018-12-30 16:20:22  16:20:22    5.0        88891   Male         Web   
51286  2018-12-30 11:52:02  11:52:02    7.0        74259   Male         Web   
51287  2018-12-30 15:06:46  15:06:46    5.0        62596   Male         Web   
51288  2018-12-30 15:36:49  15:36:49    1.0        77349   Male         Web   
51289  2018-12-30 13:49:02  13:49:02    7.0        79637   Male         Web   

      Customer_Login_type    Product_Category                 Product  Sales  \
0                  Member  Auto & Accessories       Car Media Players  140.0   
1                  Member  Auto & Accessories  Car Pillow & Neck Rest  231.0   
2                  Member  Auto & Accessories            Car Speakers  211.0   
3                  Member  Auto & Accessories                    Tyre  250.0   
4                  Member  Auto & Accessories                    Tyre  250.0   
...                   ...                 ...                     ...    ...   
51285              Member    Home & Furniture                   Sofas   67.0   
51286              Member    Home & Furniture                Curtains   34.0   
51287              Member    Home & Furniture               Umbrellas   70.0   
51288              Member    Home & Furniture         Dinner Crockery  133.0   
51289              Member    Home & Furniture             Sofa Covers  216.0   

       Quantity  Discount  Profit  Shipping_Cost Order_Priority Payment_method  
0           4.0       0.3    43.2            4.3         Medium       e_wallet  
1           5.0       0.1   139.5           13.9           High    money_order  
2           5.0       0.1   120.5           12.0           High    credit_card  
3           4.0       0.2   150.0           15.0       Critical    credit_card  
4           1.0       0.1   165.0           16.5           High    credit_card  
...         ...       ...     ...            ...            ...            ...  
51285       1.0       0.4    33.5            3.4         Medium    credit_card  
51286       5.0       0.3     6.8            0.7         Medium    credit_card  
51287       1.0       0.2    35.0            3.5         Medium    money_order  
51288       1.0       0.3    49.0            4.9         Medium    credit_card  
51289       5.0       0.3   103.6           10.4         Medium    money_order  

[51290 rows x 16 columns]

In [6]:
df["Order_Date"] = pd.to_datetime(df["Order_Date"], format="%Y-%m-%d %H:%M:%S")

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51290 entries, 0 to 51289
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   Order_Date           51290 non-null  datetime64[ns]
 1   Time                 51290 non-null  object        
 2   Aging                51289 non-null  float64       
 3   Customer_Id          51290 non-null  int64         
 4   Gender               51290 non-null  object        
 5   Device_Type          51290 non-null  object        
 6   Customer_Login_type  51290 non-null  object        
 7   Product_Category     51290 non-null  object        
 8   Product              51290 non-null  object        
 9   Sales                51289 non-null  float64       
 10  Quantity             51288 non-null  float64       
 11  Discount             51289 non-null  float64       
 12  Profit               51290 non-null  float64       
 13  Shipping_Cost        51289 non-

In [8]:
df = df.drop("Time", axis=1)
df.head()

Order_Date  Aging  Customer_Id Gender Device_Type  \
0 2018-01-01 10:11:40    5.0        29317   Male         Web   
1 2018-01-01 22:30:44    7.0        42270   Male         Web   
2 2018-01-01 21:55:31   10.0        14563   Male         Web   
3 2018-01-01 13:57:15    9.0        58601   Male         Web   
4 2018-01-01 15:17:41    2.0        48342   Male         Web   

  Customer_Login_type    Product_Category                 Product  Sales  \
0              Member  Auto & Accessories       Car Media Players  140.0   
1              Member  Auto & Accessories  Car Pillow & Neck Rest  231.0   
2              Member  Auto & Accessories            Car Speakers  211.0   
3              Member  Auto & Accessories                    Tyre  250.0   
4              Member  Auto & Accessories                    Tyre  250.0   

   Quantity  Discount  Profit  Shipping_Cost Order_Priority Payment_method  
0       4.0       0.3    43.2            4.3         Medium       e_wallet  
1       5.0       0.1   139.5           13.9           High    money_order  
2       5.0       0.1   120.5           12.0           High    credit_card  
3       4.0       0.2   150.0           15.0       Critical    credit_card  
4       1.0       0.1   165.0           16.5           High    credit_card

In [9]:
df["sales_amount"] = df["Sales"] * df["Quantity"]
df.head()

Order_Date  Aging  Customer_Id Gender Device_Type  \
0 2018-01-01 10:11:40    5.0        29317   Male         Web   
1 2018-01-01 22:30:44    7.0        42270   Male         Web   
2 2018-01-01 21:55:31   10.0        14563   Male         Web   
3 2018-01-01 13:57:15    9.0        58601   Male         Web   
4 2018-01-01 15:17:41    2.0        48342   Male         Web   

  Customer_Login_type    Product_Category                 Product  Sales  \
0              Member  Auto & Accessories       Car Media Players  140.0   
1              Member  Auto & Accessories  Car Pillow & Neck Rest  231.0   
2              Member  Auto & Accessories            Car Speakers  211.0   
3              Member  Auto & Accessories                    Tyre  250.0   
4              Member  Auto & Accessories                    Tyre  250.0   

   Quantity  Discount  Profit  Shipping_Cost Order_Priority Payment_method  \
0       4.0       0.3    43.2            4.3         Medium       e_wallet   
1       5.0       0.1   139.5           13.9           High    money_order   
2       5.0       0.1   120.5           12.0           High    credit_card   
3       4.0       0.2   150.0           15.0       Critical    credit_card   
4       1.0       0.1   165.0           16.5           High    credit_card   

   sales_amount  
0         560.0  
1        1155.0  
2        1055.0  
3        1000.0  
4         250.0

In [10]:
df["Gender"] = df["Gender"].replace({"Male": "M", "Female": "F"})

In [11]:
df["Gender"].unique()

array(['M', 'F'], dtype=object)

In [12]:
df.columns = df.columns.str.lower()
df.head()

order_date  aging  customer_id gender device_type  \
0 2018-01-01 10:11:40    5.0        29317      M         Web   
1 2018-01-01 22:30:44    7.0        42270      M         Web   
2 2018-01-01 21:55:31   10.0        14563      M         Web   
3 2018-01-01 13:57:15    9.0        58601      M         Web   
4 2018-01-01 15:17:41    2.0        48342      M         Web   

  customer_login_type    product_category                 product  sales  \
0              Member  Auto & Accessories       Car Media Players  140.0   
1              Member  Auto & Accessories  Car Pillow & Neck Rest  231.0   
2              Member  Auto & Accessories            Car Speakers  211.0   
3              Member  Auto & Accessories                    Tyre  250.0   
4              Member  Auto & Accessories                    Tyre  250.0   

   quantity  discount  profit  shipping_cost order_priority payment_method  \
0       4.0       0.3    43.2            4.3         Medium       e_wallet   
1       5.0       0.1   139.5           13.9           High    money_order   
2       5.0       0.1   120.5           12.0           High    credit_card   
3       4.0       0.2   150.0           15.0       Critical    credit_card   
4       1.0       0.1   165.0           16.5           High    credit_card   

   sales_amount  
0         560.0  
1        1155.0  
2        1055.0  
3        1000.0  
4         250.0

# **Описова аналітика**

In [13]:
max_value = df["sales_amount"].max()
max_value

1250.0

In [14]:
min_value = df["sales_amount"].min()
min_value

33.0

In [15]:
sum_value = df["sales_amount"].sum()
sum_value

np.float64(19635960.0)

In [16]:
mean_value = df["sales_amount"].mean()
mean_value

np.float64(382.86427359759784)

In [17]:
median_value = df["sales_amount"].median()
median_value

248.0

# **a**


* максимальна сума продажу – 1250.0
* мінімальна сума продажу – 33.0
* сума продажу – 19635960.0
* середнє – 382.86427359759784
* медіана – 248.0

# **b**

In [18]:
df["product_category"].unique()

array(['Auto & Accessories', 'Fashion', 'Electronic', 'Home & Furniture'],
      dtype=object)

In [19]:
df["product_category"].nunique()

4

In [20]:
category_stats = df.groupby("product_category").agg(
    orders_count=("product_category", "count"),
    total_quantity=("quantity", "sum"),
    total_sales=("sales_amount", "sum")
).reset_index()

print(category_stats)

     product_category  orders_count  total_quantity  total_sales
0  Auto & Accessories          7505         17593.0    2576442.0
1          Electronic          2701          5951.0     868178.0
2             Fashion         25646         66639.0   11298885.0
3    Home & Furniture         15438         38190.0    4892455.0


* Скільки категорій продукції продає компанія – 4
* Яка популярність кожної категорії?
  *  ***Auto & Accessories:*** кількість замовлень - 7505; кількість проданих одиниць - 17593.0; на суму –  2576442.0;
  *  ***Electronic:***  кількість замовлень - 2701; кількість проданих одиниць - 5951.0; на суму – 868178.0;
  * ***Fashion:*** кількість замовлень - 25646; кількість проданих одиниць -          66639.0; на суму – 11298885.0;
  * ***Home & Furniture:*** кількість замовлень -  15438; кількість проданих одиниць -  38190.0;  на суму – 4892455.0.

# **c**

In [21]:
df.head()

order_date  aging  customer_id gender device_type  \
0 2018-01-01 10:11:40    5.0        29317      M         Web   
1 2018-01-01 22:30:44    7.0        42270      M         Web   
2 2018-01-01 21:55:31   10.0        14563      M         Web   
3 2018-01-01 13:57:15    9.0        58601      M         Web   
4 2018-01-01 15:17:41    2.0        48342      M         Web   

  customer_login_type    product_category                 product  sales  \
0              Member  Auto & Accessories       Car Media Players  140.0   
1              Member  Auto & Accessories  Car Pillow & Neck Rest  231.0   
2              Member  Auto & Accessories            Car Speakers  211.0   
3              Member  Auto & Accessories                    Tyre  250.0   
4              Member  Auto & Accessories                    Tyre  250.0   

   quantity  discount  profit  shipping_cost order_priority payment_method  \
0       4.0       0.3    43.2            4.3         Medium       e_wallet   
1       5.0       0.1   139.5           13.9           High    money_order   
2       5.0       0.1   120.5           12.0           High    credit_card   
3       4.0       0.2   150.0           15.0       Critical    credit_card   
4       1.0       0.1   165.0           16.5           High    credit_card   

   sales_amount  
0         560.0  
1        1155.0  
2        1055.0  
3        1000.0  
4         250.0

In [22]:
top_5 = df.sort_values(by=['aging'], axis=0, ascending=False).iloc[:5]
top_5

order_date  aging  customer_id gender device_type  \
20447 2018-06-27 00:34:10   10.5        52771      M         Web   
39248 2018-10-22 16:47:28   10.0        80731      M         Web   
15932 2018-05-23 14:00:46   10.0        88594      F         Web   
35261 2018-09-28 23:17:03   10.0        40566      F         Web   
48839 2018-12-16 20:00:07   10.0        86488      M         Web   

      customer_login_type    product_category          product  sales  \
20447              Member  Auto & Accessories  Car Seat Covers  114.0   
39248              Member    Home & Furniture      Sofa Covers  216.0   
15932              Member    Home & Furniture           Towels  228.0   
35261              Member             Fashion    Running Shoes  224.0   
48839              Member    Home & Furniture       Bed Sheets  211.0   

       quantity  discount  profit  shipping_cost order_priority  \
20447       1.0       0.3    24.9            2.5           High   
39248       3.0       0.4   110.1           11.0           High   
15932       3.0       0.5   113.8           11.4           High   
35261       5.0       0.4    99.2            9.9       Critical   
48839       4.0       0.4    97.2            9.7         Medium   

      payment_method  sales_amount  
20447    credit_card         114.0  
39248    money_order         648.0  
15932    credit_card         684.0  
35261    credit_card        1120.0  
48839    credit_card         844.0

In [23]:
print(top_5["product"].tolist())

['Car Seat Covers', 'Sofa Covers', 'Towels', 'Running Shoes', 'Bed Sheets']


***топ-5 популярних продуктів***:

  💺'Car Seat Covers'

  👞'Formal Shoes'

  🥋'Sports Wear'

  ⏱️'Titak watch'
  
  🧖🏻‍♀️ 'Towels'

# **d**

In [24]:
customer_gender = df.groupby("gender").agg(
   quantity=("gender", "count"),
).reset_index()

print(customer_gender)

  gender  quantity
0      F     23152
1      M     28138



***Чоловіки купують частіше, ніж жінки.***




# **e**

In [25]:
device = df.groupby("device_type").agg(
   quantity=("device_type", "count"),
).reset_index()

print(device)

  device_type  quantity
0      Mobile      3658
1         Web     47632


 ***Найбільш популярний метод оплати – WEB***